LgiNetwork/Network.cpp

```diff
@@ -0,0 +1,995 @@
+        Network* Network::_instance = nullptr;
+
+        Network::Network() : PluginHost::JSONRPC()
+        {
+            Network::_instance = this;
+            m_isPluginInited = false;
+
+            // Quirk
+            Register("getQuirks", &Network::getQuirks, this);
+
+            // Network_API_Version_1
+            Register("getInterfaces", &Network::getInterfaces, this);
+            Register("isInterfaceEnabled", &Network::isInterfaceEnabled, this);
+            Register("setInterfaceEnabled", &Network::setInterfaceEnabled, this);
+            Register("getDefaultInterface", &Network::getDefaultInterface, this);
+            Register("setDefaultInterface", &Network::setDefaultInterface, this);
+
+            Register("getStbIp", &Network::getStbIp, this);
+
+            Register("trace", &Network::trace, this);
+            Register("traceNamedEndpoint", &Network::traceNamedEndpoint, this);
+
+            Register("getNamedEndpoints", &Network::getNamedEndpoints, this);
+
+            Register("ping",              &Network::ping, this);
+            Register("pingNamedEndpoint", &Network::pingNamedEndpoint, this);
+
+            Register("setIPSettings", &Network::setIPSettings, this);
+            Register("getIPSettings", &Network::getIPSettings, this);
+
+            Register("getSTBIPFamily", &Network::getSTBIPFamily, this);
+            Register("isConnectedToInternet", &Network::isConnectedToInternet, this);
+            Register("setConnectivityTestEndpoints", &Network::setConnectivityTestEndpoints, this);
+
+            Register("getPublicIP", &Network::getPublicIP, this);
+            Register("setStunEndPoint", &Network::setStunEndPoint, this);
+
+            m_defaultInterface = "";
+            m_gatewayInterface = "";
+
+            m_netUtils.InitialiseNetUtils();
+            m_stunEndPoint = "stun.l.google.com";
+            m_stunPort = 19302;
+            m_stunBindTimeout = 30;
+            m_stunCacheTimeout = 0;
+            m_stunSync = true;
+        }
+
+        Network::~Network()
+        {
+        }
+
+        const string Network::Initialize(PluginHost::IShell* /* service */)
+        {
+            string msg;
+            int rcode = m_NetworkClient.Run();
+            if (rcode)
+            {
+                LOGERR("Failed to initialize DBus Network Client (%d)", rcode);
+                msg = "Failed to initialize DBus Network Client";
+                m_isPluginInited = false;
+            }
+            else
+            {
+                m_isPluginInited = true;
+                m_NetworkClient.onStatusChangeEvent = StatusChangeEvent;
+                m_NetworkClient.onNetworkingEvent = NetworkingEvent;
+                LOGINFO("Succeeded initializing LGI DBus Network Client");
+            }
+            return msg;
+        }
+
+        void Network::Deinitialize(PluginHost::IShell* /* service */)
+        {
+            m_isPluginInited = false;
```

Comment:
IMHO we need some deinitialization for: m_NetworkClient.Run(); ?

see the following in logs while plugin Initialization 2nd time:
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: [8] INFO [lginetwork_client.cpp:288] Run: user_data: 0x133e1fc
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-WARNING **: 12:33:42.781: cannot register existing type 'Networkconfig1Proxy'
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-WARNING **: 12:33:42.782: cannot add private field to invalid (non-instantiatable) type '<invalid>'
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-WARNING **: 12:33:42.782: cannot register existing type 'Networkconfig1'
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-CRITICAL **: 12:33:42.782: g_type_interface_add_prerequisite: assertion 'G_TYPE_IS_INTERFACE (interface_type)' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-CRITICAL **: 12:33:42.782: g_once_init_leave: assertion 'result != 0' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GObject-CRITICAL **: 12:33:42.782: g_type_add_interface_static: assertion 'G_TYPE_IS_INSTANTIATABLE (instance_type)' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-CRITICAL **: 12:33:42.782: g_once_init_leave: assertion 'result != 0' failed
Jan 16 13:33:42 E0B7B1-APPSTB-301000004400 wpeframework.sh[4592]: (process:2): GLib-GIO-CRITICAL **: 12:33:42.782: g_initable_new_valist: assertion 'G_TYPE_IS_INITABLE (object_type)' failed